mvebu: backport mvneta and comphy from linux 5.x

ttocsr opened the pull request:
These patches backport support for the ARMADA 3700 COMPHY driver.
Also backported is the mvneta driver.  This will allow switching
the SGMII speed using SMC calls.  To support this you must update
the firmware using Marvells 18.12 version (this has now been
upstreamed).  The mvneta driver allows 2500basex and 2500baset.

tmn505 commented:
Hi,
I tested this on ESPRESSObin with factory bootloader and there seems to be no regressions. But this touches `phy-mvebu-cp110-comphy` and `mvneta` so other devices would need testing.
@v-vid or @damirsamardzic could You test this on one of the boards from cortexa72 subtarget?
@KanjiMonster could You test this on ClearFog?
The rest are only formal issues, proper formatting of commit message, SoB line and since these are all backports, please recreate the patches with `git format-patch -1 <commit hash>` from Linus tree (You can delete from them chunks of unrelated changes). For last after putting all patches in place do `make target/linux/{clean,refresh} V=s`.

P.S.
Some guidelines for submitting patches https://openwrt.org/submitting-patches, if I omitted something.

ttocsr commented:
Added SoB. I recreated the patches as requested.
If you update the bootloader on the espressobin and update the espressobin dts you can use comphy there.
Thanks.

tmn505 reviewed (commented):
Small nitpicks regarding cosmetics in the commit message: remove `linux:` label or swap places with `mvebu:`, because those changes are unrelated to other targets, and also insert blank line before SoB line.

Regarding strict patches handling. Don't worry too much about the amount of added patches, since those are backports. This will speed up the process of review and will increase likelihood of merging this PR by core developers (I don't have this privilege).

> If you update the bootloader on the espressobin and update the espressobin dts you can use comphy there.

Last time I checked this, A3700-utils and ATF from Marvell were very picky about used toolchain. Currently I don't have much time to check this feature, I will only check for regressions, but will definitely try it in the future.

ttocsr committed:
mvebu: backport mvneta and comphy from linux 5.x

These patches backport support for the ARMADA 3700 COMPHY driver.
Also backported is the mvneta driver.  This will allow switching
the SGMII speed using SMC calls.  To support this you must update
the firmware using Marvells 18.12 version (this has now been
upstreamed).  The mvneta driver allows 2500basex and 2500baset.

Signed-off-by: Scott Roberts <ttocsr@gmail.com>

ttocsr commented on target/linux/mvebu/patches-4.19/411-net-phy-enumerate-phy-modes.patch:
```diff
@@ -0,0 +1,31 @@
+From c2a90025ad09d830c8d8ae69f485eac6aaaa2472 Mon Sep 17 00:00:00 2001
```
I broke this out and removed PHY_MODE_MIPI_DPHY

ttocsr commented on target/linux/mvebu/patches-4.19/531-arm64-dts-armada-37xx_dtsi_comphy.patch:
```diff
@@ -0,0 +1,173 @@
+From 620cfb31bad4c5be7d9250f1e47a592750fc364b Mon Sep 17 00:00:00 2001
```
Broken out as suggested

ttocsr commented on target/linux/mvebu/patches-4.19/532-mvneta-backport-5.x.patch:
```diff
@@ -0,0 +1,259 @@
+From da58a931f248f423f917c3a0b3c94303aa30a738 Mon Sep 17 00:00:00 2001
```
I moved them all to the end since there are quite a few patches that were combined.

tmn505 commented:
Also separate the title from message body with blank line in Your commit message, it was good in 2nd revision.
Other than that, from my POV this looks good.

hauke commented:
Thank you for your patch, it was applied to master.
